core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -133,16 +142,31 @@ public boolean isAccessible(Map context, Object target, Member member, String pr
             return false;
         }
 
-        return Modifier.isPublic(member.getModifiers()) && isAcceptableProperty(propertyName);
+        return Modifier.isPublic(memberModifiers) && isAcceptableProperty(propertyName);
     }
 
-    protected boolean checkStaticMethodAccess(Member member) {
+    /**
+     * Check access for static members
+     * 
+     * Static non-field access result is a logical and of allowStaticMethodAccess and public.
+     * Static field access result is a logical and of allowStaticFieldAccess and public.
+     * Note: For non-static members, the result is always true.
+     * 
+     * @param member
+     * 
+     * @return
+     */
+    protected boolean checkStaticMemberAccess(Member member) {
         final int modifiers = member.getModifiers();
         if (Modifier.isStatic(modifiers)) {
-            if (allowStaticMethodAccess) {
-                LOG.debug("Support for accessing static methods [member: {}] is deprecated!", member);
+            if (member instanceof Field) {
+                return allowStaticFieldAccess && Modifier.isPublic(modifiers);
```

Comment:
Hello @lukaszlenart .
I had considered not performing the check, but figured that even though the "catch-all" public check happens at line 145 that both the checks in `checkStaticMemberAccess()` should remain for two reasons:
1) It is needed to continue to provide the Log Warn when access is denied due to a member/field being non-public.
2) As a protected member, it could be called independently (e.g. from a descendant) so the public access check ensures consistency in return value.

Since `line 145`  provides a "must-be-public" check at the end, I'm wondering if the whole sequence should be changed, adding a new `checkPublicMemberAccess()` check between `checkEnumAccess()` and `checkStaticMemberAccess()`.  That would have a few advantages:
1) Minimize extra checks for non-public members (by determining a false result early-on)
2) Allow `checkStaticMemberAccess()` to only check the boolean flag states (as per your recommendations).
3) Allow line 145 to only perform an `isAcceptableProperty()` check (since the public check would happen earlier on).

If yourself and @aleksandr-m think that makes sense I could modify the PR to do that instead (and if it makes sense here, it could be retrofitted to PR#316 for 2.5.21).

What-do-you-think ?

Reply:
Wondering, why `isPublic` is not the very first check.?

Reply:
Hello again.

I'm thinking that `isPublic `wasn't the first check in order to (possibly) allow for non-public enum access and also the log warning for static fields (to indicate a blocked access attempt).

Perhaps @lukaszlenart  has a better sense as to why.  Without being 100% sure, I'd err on the side of caution and do the `isPublic` check after those first two checks (to avoid introducing a possible breaking behaviour) ...

Reply:
This code was manually merged from OGNL after I have thrown away `DefaultAccessMember` in OGNL 3.2.x - so changing order or so is ok, but I would first merge this PR and then do another refactorings as needed.